### PR TITLE
fix(davinci-client): small fixes found during e2e testing

### PR DESCRIPTION
**Pull request by cerebrl**
# JIRA Ticket

[Jira ticket](https://bugster.forgerock.org/jira/browse/SDKS-3393)

# Description

Added some fixes to DaVinci client found in manual e2e tests and also added test app to repo under `/e2e` directory. 

# Type of Change

Please Delete options that are not relevant

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Pure manual testing with shared tenant and flows

# Definition of Done

Check all that apply

- [x] Acceptance criteria is met.
- [x] All tasks listed in the user story have been completed.
- [x] Coded to standards.
- [ ] Code peer-reviewed.
- [x] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: e2e/davinci-app/components/password.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { SingleValueCollector } from '../../packages/davinci/types.ts';
```
Comment (ryanbas21) [inline e2e/davinci-app/components/password.ts:1]: why do we need the .ts file extensions? Are we not exporting an index file from the davinci package?

File: e2e/davinci-app/main.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import passwordComponent from './components/password.ts';
+import submitButtonComponent from './components/submit-button.ts';
+import protect from './components/protect.ts';
+import flowLinkComponent from './components/flow-link.ts';
+import socialLoginButtonComponent from './components/social-login-button.ts';
+
+/**
+ * Import types for the DaVinci client
+ */
+import type { SingleValueCollector, ActionCollector } from '@forgerock/davinci-client/types.ts';
```
Comment (ryanbas21) [inline e2e/davinci-app/main.ts:16]: i'm confused by this import, the `@forgerock/davinci-client` is used here but other places we are using relative imports?

File: packages/davinci-client/package.json [hunk 3]
```diff
@@ -8,5 +8,12 @@
   "dependencies": {
     "@forgerock/javascript-sdk": "*",
     "@reduxjs/toolkit": "^2.2.5"
+  },
+  "exports": {
+    ".": {
+      "import": "./src/index.ts",
```
Comment (ryanbas21) [inline packages/davinci-client/package.json:14]: I think these should be js files and relative to the dist directory, so if this package.json is not in the `dist` directory we would either want to a) copy the package.json file into the dist directory or b) point the package.json exports to the dist/src/index.js file

Comment (ryanbas21) [reply packages/davinci-client/package.json:14]: unless ofc, we need to edit these after the build process, i believe i do do that or used to do that elsewhere

File: packages/davinci-client/src/lib/collector.types.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ export interface ActionCollector {
   };
 }
 
-export interface FlowCollector extends ActionCollector {}
-export interface SocialLoginCollector extends ActionCollector {}
-export interface PasswordCollector extends SingleValueCollector {}
-export interface TextCollector extends SingleValueCollector {}
+export type FlowCollector = ActionCollector;
+export type SocialLoginCollector = ActionCollector;
```
Comment (ryanbas21) [inline packages/davinci-client/src/lib/collector.types.ts:49]: Are these aliases going to have a more significant difference going forward? Like SocialLoginCollector and FlowCollector, will they have differences in the type? Or are we going to always have these aliases for the same type?

File: packages/davinci-client/src/lib/config.slice.ts [hunk 5]
```diff
@@ -37,8 +37,7 @@ export const configSlice = createSlice({
      */
     set(state, action: PayloadAction<DaVinciConfig>) {
       state.clientId = action.payload.clientId || '';
-      state.redirectUri =
-        action.payload.redirectUri || `${location.origin}/handle-redirect`;
+      state.redirectUri = action.payload.redirectUri || `${location.origin}/handle-redirect`;
```
Comment (ryanbas21) [inline packages/davinci-client/src/lib/config.slice.ts:40]: This second half the of the `||`, is `handle-redirect` expected to be there on the client? Or should this not be apart of the published code?

File: e2e/davinci-app/components/password.ts (hunk 1 above)
Comment (cerebrl) [reply e2e/davinci-app/components/password.ts:1]: I'll take a look. I think this would auto-added.

File: e2e/davinci-app/main.ts (hunk 2 above)
Comment (cerebrl) [reply e2e/davinci-app/main.ts:16]: I think I just partially corrected some imports as I moved the code over to this repo. I'll make sure they're consistent.

File: packages/davinci-client/src/lib/collector.types.ts (hunk 4 above)
Comment (cerebrl) [reply packages/davinci-client/src/lib/collector.types.ts:49]: I'm expecting these types to evolve differently as they mature. Right now, they just happen to be the same. Social Login isn't fully developed yet and will likely have additional configuration for styling and such.

File: packages/davinci-client/src/lib/config.slice.ts (hunk 5 above)
Comment (cerebrl) [reply packages/davinci-client/src/lib/config.slice.ts:40]: Yes, the fallback is just a recommended default. The customer will need a dedicated route/page for this, or they provide a their own redirect URL.

**Comment by cerebrl** [comment]
Closing in favor of this other PR: https://github.com/ForgeRock/forgerock-javascript-sdk/pull/462

**Pull request closed without merging**